### PR TITLE
Create config.ts, replace all process.env with config, add guideline.md

### PR DESCRIPTION
**Pull request by Corei13**

### CODE REVIEW

File: apps/web/config.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import getConfig from "next/config";
+
+const getEnv = (key: string, defaultValue: unknown = ""): string => {
+  if (key.includes("public")) {
```
Comment (Corei13) [inline apps/web/config.ts:4]:
```ts
const { publicRuntimeConfig } = getConfig()
if (key.startsWith('public') && key in publicRuntimeConfig) {  return publicRuntimeConfig[key] }
if (key in process.env) { return process.env[key] }

return String(defaultValue)
```

File: apps/web/config.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+      supabaseUrl: getEnv("publicSupabaseUrl"),
+      supabaseKey: getEnv("publicSupabaseKey"),
+      sentryDsn: getEnv("publicSentryDsn"),
+      replicache: {
+        licenceKey: getEnv("publicReplicacheLicenceKey"),
+      },
+    },
+  },
+  googleClientSecret: getEnv("GOOGLE_CLIENT_SECRET"),
+  sentryDsn: getEnv("SENTRY_DSN"),
```
Comment (Corei13) [inline apps/web/config.ts:24]: let's remove `sentryDsn` since we already have this inside public

File: apps/web/config.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+const getEnv = (key: string, defaultValue: unknown = ""): string => {
+  if (key.includes("public")) {
+    return getConfig().publicRuntimeConfig[key] ?? String(defaultValue);
+  } else {
+    return process.env[key] ?? String(defaultValue);
+  }
+};
+
+const config = {
```
Comment (Corei13) [inline apps/web/config.ts:11]: let's format in this way
```ts
{
  secret: {
      database: { url },
      google: { clientSecret }
  },
  public: {
    supabse: { url, key },
    sentry: { dsn },
    replicache: { licenceKey }
  }
}
```

**Review by Corei13 — COMMENTED** [review]
looks good, a few feedback on naming and structure

File: apps/web/lib/__experimental__/replicache-context.tsx [hunk 4]
```diff
@@ -25,7 +26,8 @@ const hoistReplicache = (
   if (currentWindow.parent === currentWindow.self) {
     currentWindow.replicache = new Replicache({
       // See https://doc.replicache.dev/licensing for how to get a license key.
-      licenseKey: process.env.NEXT_PUBLIC_REPLICACHE_LICENSE_KEY!,
+      licenseKey: config.next.public.replicache.licenceKey!,
```
Comment (ranit43) [inline apps/web/lib/__experimental__/replicache-context.tsx:29]: Can we do
```suggestion
      licenseKey: config.next.public.replicache?.licenceKey,
```
instead?

File: apps/web/config.ts (hunk 2 above)
Comment (shubasishdas) [reply apps/web/config.ts:24]: @Corei13 
we had 2 types of SENTRY_DSN ( public & non-public) in [apps/web/sentry.client.config.js](https://github.com/Corei13/reframe/pull/54/files#diff-2c585a3ea77d2a504ae9fedc95e113780d6cba6f797ee51e541031c8484390d0) & [apps/web/sentry.server.config.js](https://github.com/Corei13/reframe/pull/54/files#diff-a11cfaa8c6769b8e40a1f49941371c518b1c5ca2106eb96bf22ed678fe67fea9) 

 `const SENTRY_DSN = process.env.SENTRY_DSN || process.env.NEXT_PUBLIC_SENTRY_DSN;`
 
 
Will I keep it or remove?

File: apps/web/config.ts (hunk 1 above)
Comment (shubasishdas) [reply apps/web/config.ts:4]: @Corei13 should we include `process.env[key]` in logic cause without that value can be null.
 
 ```
if (key in process.env && process.env[key]) {
    return process.env[key] as string;
  }
```

File: apps/web/lib/__experimental__/replicache-context.tsx (hunk 4 above)
Comment (shubasishdas) [reply apps/web/lib/__experimental__/replicache-context.tsx:29]: okay, doing that. 

If we're not 100% sure about having the specific object key, we shouldn't use `!`

**Pull request merged**